File: BackendDashboard/src/main/java/ru/dashboard/backenddashboard/search/TaskSearchValues.java
package ru.dashboard.backenddashboard.search;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ru.dashboard.backenddashboard.entity.Category;
import ru.dashboard.backenddashboard.entity.Priority;
import java.util.Date;

@NoArgsConstructor
@Getter
@Setter
@AllArgsConstructor
public class TaskSearchValues {
    protected String title;
    protected Integer completed;
    protected Date date;
    protected Long priorityId;
    protected Long categoryId;

    protected Integer pageNumber;

    protected Integer pageSize;
    protected String sortColumn;
    protected String sortDirection;
}

File: Frontend/src/app/services/DataAPI/search/TaskSearchValues.ts
export class TaskSearchValues {
    public  title:String='';
    public  completed:number=null;
    public  date:Date=null;
    public  priorityId:number=null;
    public  categoryId:number=null;

    public  pageNumber:number=0;

    public  pageSize:number=5;
    public  sortColumn:String='title';
    public  sortDirection:String='asc';
}
